### ISSUE TITLE
Feature Request: Add Support for BGP Communities in Feed Processing

### ISSUE THREAD

**Issue by soakes** [ebarkie/blackhole-threats#6]
Hi @ebarkie,

First off, thank you for creating and maintaining this project! It's a fantastic tool for managing threat intelligence feeds and integrating them with GoBGP for blackholing malicious routes.

I'd like to request a feature enhancement to add support for BGP communities in the feed processing. Currently, the software supports multiple feeds, which is great, but it doesn't allow for the use of BGP communities to tag or differentiate routes based on their source feed. Adding this functionality would enable users to:

1. **Differentiate Feeds:** Assign specific communities to routes based on the feed they originate from, making it easier to identify and manage them downstream.
1. **Policy Control:** Apply routing policies or filters based on the community tags, allowing for more granular control over how routes are handled.
1. **Scalability:** Simplify the management of multiple feeds by leveraging community-based tagging instead of relying solely on feed-specific configurations.

## Proposed Implementation:

- Add an optional configuration parameter for each feed to specify a BGP community (e.g., `64512:100`)
- When processing a feed, assign the specified community to all routes derived from that feed
- Ensure that the community tags are propagated to the BGP daemon and included in the advertised routes

## Example Configuration:

```
feeds:  
  - name: "abuse.ch"  
    url: "https://example.com/abusech-feed"  
    community: "64512:100"  
  - name: "spamhaus"  
    url: "https://example.com/spamhaus-feed"  
    community: "64512:200"  
```

## Benefits:

This feature would greatly enhance the flexibility and usability of the tool, especially for users managing a large number of feeds or complex routing environments. It would also align with best practices for BGP route management by leveraging communities for route tagging and policy enforcement.

Please let me know if this is something you'd consider implementing. I'd be happy to provide further details or assist with testing if needed.

Thanks again for your hard work on this project!

**Comment by ebarkie** [comment]
Hey @soakes, apologies for not responding sooner.

Thanks for the detailed and very well thought out feature request!  I'm not actively using this any more but it's a great idea and I *think* it would be pretty easy to implement, so I'd be willing to give it a shot.

The only tradeoff I can think of is that we would need to summarize the routes per-community, but that's a conscious choice someone would be making by using multiple communities.

I'll try to take a stab at this some time over the next few days..

**Comment by soakes** [comment]
> Hey @soakes, apologies for not responding sooner.
> 
> Thanks for the detailed and very well thought out feature request! I'm not actively using this any more but it's a great idea and I _think_ it would be pretty easy to implement, so I'd be willing to give it a shot.
> 
> The only tradeoff I can think of is that we would need to summarize the routes per-community, but that's a conscious choice someone would be making by using multiple communities.
> 
> I'll try to take a stab at this some time over the next few days..

Hi @ebarkie,

No worries at all about the delay in responding, and thank you so much for taking the time to consider my feature request!

I really appreciate your willingness to give it a shot, especially since you're not actively using the tool anymore. It's such a useful tool, and it's been incredibly helpful for me, so I just wanted to say a big thank you for all the work you've put into it.

The tradeoff you mentioned about summarizing routes per-community makes total sense, and I completely understand that it would be a conscious choice for users. I'm sure whatever approach you take will work great!

Thanks again for your time and effort, and I hope you have a wonderful holiday season. Wishing you a Merry Christmas and all the best for the New Year! 🎄

**Comment by ebarkie** [comment]
Released as v0.10.0!
